src/main/java/com/infrastructure/PostgressImpl.java

```diff
@@ -0,0 +1,36 @@
+package com.infrastructure;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.stereotype.Repository;
+
+import com.javapractice.gateways.ITodoRepository;
+
+@Repository
```

Comment:
This @Repository decorator allows this repository to enter the inversion of control (IOC) container. It basically just means it's now injectable.